k2/csrc/fsa_equivalent.cc

```diff
@@ -0,0 +1,82 @@
+// k2/csrc/fsa_equivalent.cc
+
+// Copyright (c)  2020 Haowen Qiu
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/csrc/fsa_equivalent.h"
+
+#include <algorithm>
+#include <random>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "k2/csrc/fsa.h"
+// #include "k2/csrc/fsa_algo.h"
+#include "k2/csrc/properties.h"
+
+namespace k2 {
+
+bool RandomPath(const Fsa &a, Fsa *b, std::vector<int32_t> *state_map) {
+  if (IsEmpty(a) || b == nullptr) return false;
+  if (!IsConnected(a)) {
+    Fsa connected_fsa;
+    // TODO(haowen): uncomment this
+    // Connect(a, &connected_fsa);
+    return RandomPath(connected_fsa, b, state_map);
+  }
+
+  int32_t num_states = a.NumStates();
+  std::vector<int32_t> state_map_b2a;
+  std::vector<int32_t> state_map_a2b(num_states, -1);
+  // `visited_arcs[i]` stores `arcs` leaving from state `i`
+  std::vector<std::unordered_set<Arc, ArcHash>> visited_arcs;
```

Comment:
```cpp
using ArcSet = std::unordered_set<Arc, ArcHash>;
std::vector<ArcSet> visited_arcs;
....

visited_arcs.push_bac(ArcSet{});
```

Reply:
As there is only two occurrences of  `std::unordered_set<Arc, ArcHash>`, I think using a name alias is not necessary.